@types/react-native has wrong typings for FlatList> ItemSeparatorComponent

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @....
I'm currently using `FlatList` on react-native `0.47.1` 

The type definition is technically incorrect as it does not account for `React.Element<any>:`

`ItemSeparatorComponent?: React.ComponentClass<any> | null`

As a result, you will see errors like this when trying to pass in an element as a ItemSeparatorComponent:

`error TS2352: Type 'Element' cannot be converted to type 'ComponentClass<any>'. Type 'Element' provides no match for the signature 'new (props?: any, context?: any): Component<any, ComponentState>'`

From what've tried, this fixes the issue (works the same way as ListEmptyComponent, Header and Footer):
`ItemSeparatorComponent?: React.ComponentClass<any> | React.ReactElement<any> | (() => React.ReactElement<any>) | null`

@alloy , @timwangdev , @jeremistadler  would you mind checking it out?

--- a/types/react-native/index.d.ts
+++ b/types/react-native/index.d.ts
@@ -3491,7 +3491,7 @@ export interface FlatListProperties<ItemT> extends VirtualizedListProperties<Ite
     /**
      * Rendered in between each item, but not at the top or bottom
      */
-    ItemSeparatorComponent?: React.ComponentClass<any> | React.ReactElement<any> | (() => React.ReactElement<any>) | null
+    ItemSeparatorComponent?: React.ComponentType<any> | null
 
     /**
      * Rendered when the list is empty.

--- a/types/react-native/index.d.ts
+++ b/types/react-native/index.d.ts
@@ -3491,7 +3491,7 @@ export interface FlatListProperties<ItemT> extends VirtualizedListProperties<Ite
     /**
      * Rendered in between each item, but not at the top or bottom
      */
-    ItemSeparatorComponent?: React.ComponentClass<any> | null
+    ItemSeparatorComponent?: React.ComponentClass<any> | React.ReactElement<any> | (() => React.ReactElement<any>) | null
 
     /**
      * Rendered when the list is empty.

--- a/types/react-native/test/index.tsx
+++ b/types/react-native/test/index.tsx
@@ -208,11 +208,22 @@ InteractionManager.runAfterInteractions(() => {
 }).then(() => 'done')
 
 export class FlatListTest extends React.Component<FlatListProperties<number>, {}> {
+    _renderItem = (rowData: any) => {
+        return (
+            <View>
+                <Text> {rowData.item} </Text>
+            </View>
+        );
+      }
+
+    _renderSeparator= () => <View style={{height: 1, width: '100%', backgroundColor: 'gray'}} />
+
     render() {
         return (
             <FlatList
                 data={[1, 2, 3, 4, 5]}
-                renderItem={(info: { item: number }) => <View><Text>{info.item}</Text></View>}
+                renderItem={this._renderItem}
+                ItemSeparatorComponent={this._renderSeparator}
             />
         );
     }

